templates/vite/vite.config.ts

```diff
@@ -0,0 +1,14 @@
+import { defineConfig, loadEnv } from 'vite';
+import react from '@vitejs/plugin-react-swc';
+import { TanStackRouterVite } from '@tanstack/router-plugin/vite';
```

Comment:
What is this plugin for?

Reply:
It's for the router that we use - Tanstack Router. It's optional and we can remove it if we want to. It just adds a nice overlay that lets people debug the routing etc visually. I was unsure about it as well but kept it in for now

Reply:
Hmm IMO remove anything that isn't necessary.